enterprise/api/v1/views.py

```diff
@@ -577,17 +574,68 @@ def license_revoke(self, request, *args, **kwargs):
             course_id = str(course_overview.get('id'))
             enterprise_enrollment = enrollments_by_course_id.get(course_id)
             try:
-                revocation_status = self._revoke_enrollment(
+                revocation_status = self._terminate_enrollment(
                     enrollment_api_client, enterprise_enrollment, course_overview
                 )
                 revocation_results[course_id] = {'success': True, 'message': revocation_status}
+                if revocation_status != self.EnrollmentTerminationStatus.COURSE_COMPLETED:
+                    enterprise_enrollment.license.revoke()
             except EnrollmentModificationException as exc:
                 revocation_results[course_id] = {'success': False, 'message': str(exc)}
                 any_failures = True
+                enterprise_enrollment.license.revoke()
 
         status_code = status.HTTP_200_OK if not any_failures else status.HTTP_422_UNPROCESSABLE_ENTITY
         return Response(revocation_results, status=status_code)
 
+    @action(methods=['post'], detail=False)
+    @permission_required('enterprise.can_enroll_learners')
+    def licensed_enrollments_expiration(self, request):
+        """
+        Changes the mode for licensed enterprise course enrollments to the "audit" course mode,
+        or unenroll the user if no audit mode exists for each expired license uuid
+        """
+        expired_license_uuids = get_request_value(request, self.REQ_EXP_LICENSE_UUIDS_PARAM, '')
+
+        if not expired_license_uuids:
+            return Response('expired license uuids must be provided', status=status.HTTP_400_BAD_REQUEST)
```

Comment:
nit: maybe include `self.REQ_EXP_LICENSE_UUIDS_PARAM` in the response so it's clear exactly which param is needed.